pkg/policy/resolve_test.go

```diff
@@ -443,9 +445,22 @@ func TestL7WithLocalHostWildcard(t *testing.T) {
 	require.Truef(t, policy.policyMapState.Equal(&expectedEndpointPolicy.policyMapState), policy.policyMapState.diff(&expectedEndpointPolicy.policyMapState))
 	policy.policyMapState = mapState{}
 	expectedEndpointPolicy.policyMapState = mapState{}
+	// reset cached envoy http rules to avoid unnecessary diff
+	resetCachedEnvoyHTTPRules(policy)
 	require.Equal(t, &expectedEndpointPolicy, policy)
 }
 
+func resetCachedEnvoyHTTPRules(policy *EndpointPolicy) {
+	policy.L4Policy.Ingress.PortRules.ForEach(func(l4 *L4Filter) bool {
```

Comment:
Even if not currently needed for Egress, maybe reset for both?

Reply:
good point - updated